### PR TITLE
Replace explicit richCodeNavigationEnvironment in azure-pipelines-richnav.yml

### PR DESCRIPTION
**Pull request by akoeplinger**
This is managed by the arcade defaults, we should be using the "internal" environment as requested by the RichNav team.

Also update to the correct AzDO pool name.

### CODE REVIEW

**Comment by akoeplinger** [comment]
@tmat ping

File: azure-pipelines-richnav.yml [hunk 1]
```diff
@@ -30,7 +29,7 @@ stages:
       jobs:
       - job: Windows
         pool:
-            name: NetCore1ESPool-Public
+            name: NetCore-Public
             demands: ImageOverride -equals windows.vs2019.amd64.open
```
Comment (ViktorHofer) [inline azure-pipelines-richnav.yml:33]:
```suggestion
            demands: ImageOverride -equals windows.vs2022.amd64.open
```

**Comment by akoeplinger** [comment]
I've updated the images and made the config more similar to other repos.

**Review by ViktorHofer — APPROVED** [review]

**Review by tmat — APPROVED** [review]

**Pull request merged**